exercises/react/src/example.c

```diff
@@ -41,42 +43,54 @@ struct reactor *create_reactor()
    return calloc(1, sizeof(struct reactor));
 }
 
-void destroy_reactor(struct reactor *r)
+static void destroy_cell(struct cell *c)
 {
-   struct child *child = r->first;
-   while (child) {
-      struct cb *cb = child->cell->cb;
-      while (cb) {
-         struct cb *next_cb = cb->next;
-         free(cb);
-         cb = next_cb;
-      }
-      free(child->cell);
+   struct cb *cb = c->cb;
+   while (cb) {
+      struct cb *next_cb = cb->next;
+      free(cb);
+      cb = next_cb;
+   }
 
+   struct child *child = c->child;
+   while (child) {
       struct child *next = child->next;
+      if (c == child->cell->input1) {
+         // Don't double-free for a compute2 cell.
```

Comment:
let's say it's not there, so that we always call `destroy_cell(child->cell)`.

In that case, even this:

```c
   struct cell *input1 = create_input_cell(r, 1);
   struct cell *input2 = create_input_cell(r, 2);
   struct cell *output = create_compute2_cell(r, input1, input2, plus);
```

will double-free:

```
destroy_reactor calls destroy_cell(input1)
destroy_cell(input1) calls destroy_cell(output)
destroy_reactor calls destroy_cell(input2)
destroy_cell(input2) calls destroy_cell(output)
```

Everything else is correct; it just needed to avoid a second call to `destroy_cell(output)`